crowbar_framework/app/models/crowbar_service.rb

```diff
@@ -176,6 +228,18 @@ def create_proposal
     base
   end
 
+  def apply_role_pre_chef_call(old_role, role, all_nodes)
+    @logger.debug("crowbar apply_role_pre_chef_call: entering #{all_nodes.inspect}")
+    all_nodes.each do |n|
+      node = NodeObject.find_node_by_name n
+      # value of crowbar_wall["crowbar_upgrade"] indicates that the role should be executed
+      # but node state should not be changed: this is needed when reverting node state to ready
+      if (node.role?("crowbar-upgrade") || node.role?("crowbar-db-dump")) && node.crowbar_wall["crowbar_upgrade"]
```

Comment:
Metrics/LineLength: Line is too long. [113/100] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)